### PR TITLE
Fix "error reading TLD cache file" error

### PR DESCRIPTION
**Pull request by mghughes**
The cache_file argument to TLDExtract was an empty tempfile,
so pickle complained about an EOF error when loading it.
That was then logged as an "error reading TLD cache file"
error by tldextract.

Contrary to an empty cache_file, tldextract deals with a non-existent
cache_file just fine, so we don't really need that intermediate
step of creating a tempfile anyway. Use the TLDExtract `update` method
to directly update multisite's public suffix list

### CODE REVIEW

File: multisite/management/commands/update_public_suffix_list.py [hunk 1]
```diff
@@ -20,17 +20,8 @@ def handle_noargs(self, **options):
...
-
-            extract = tldextract.TLDExtract(fetch=True, cache_file=tmpname)
-            extract._get_tld_extractor()
-            self.log(
-                "Downloaded new data to {filename}".format(filename=tmpname)
-            )
-            os.rename(tmpname, filename)
-            f.delete = False    # No need to delete f any more.
-
+        extract = tldextract.TLDExtract(fetch=True, cache_file=filename)
```
Comment (pauricthelodger) [inline multisite/management/commands/update_public_suffix_list.py]: Fetch is deprecated in 1.7.5 and removed in 2. It'll default to `True` here without being added explicitly and looks to work the same without it in 2.

Comment (mghughes) [inline multisite/management/commands/update_public_suffix_list.py]: Good point. Shall I make a separate PR for supporting tldextract >=2 though?

Comment (pauricthelodger) [inline multisite/management/commands/update_public_suffix_list.py]: Oh yeah, it'd be a good idea to have a separate PR for v2 support. It was just that if you removed the kwarg now it'd still work on both so would save on having to change it again later. Up to you!
